galaxykit/utils.py

```diff
@@ -36,7 +35,7 @@ def wait_for_task(api_client, task, timeout=300, raise_on_error=False):
                 logger.error(resp["error"])
                 if raise_on_error:
                     raise TaskFailed(resp["error"])
-        except GalaxyError as e:
+        except BaseException as e:
```

Comment:
If we really have to get rid of GalaxyError and it doesn't inherit from something else we can catch like ValueError or similar, we should at least avoid ever catching bare exceptions with BaseException.

BaseException includes system/internal exceptions like KeyboardInterrupt, which can cause a process to be difficult to kill or behave otherwise badly. In the worst case, catch Exception if you really have to!

Reply:
Actually, looking at this again, shouldn't the client be raising GalaxyClientError and we should catch that? GalaxyError was never correct because we don't use ansible's client in galaxykit anyway, but we have our own, specific exceptions we _can_ catch.

Reply:
Ah definitely agree re: the use of BaseException, I hadn't thought that through very well. Updated to use GalaxyClientError!